src/contrib/mlir/pass/mlir_subgraph_extraction.cpp

```diff
@@ -0,0 +1,104 @@
+//*****************************************************************************
+// Copyright 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "mlir_subgraph_extraction.hpp"
+#include "ngraph/assertion.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/add.hpp"
+#include "ngraph/op/dot.hpp"
+#include "ngraph/op/experimental/compiled_kernel.hpp"
+#include "ngraph/op/get_output_element.hpp"
+
+using namespace ngraph::descriptor;
+using namespace ngraph::op;
+using namespace ngraph::pass;
+
+#define TI(x) std::type_index(typeid(x))
+
+bool MLIRSubgraphExtractionPass::run_on_function(std::shared_ptr<Function> func)
+{
+    // Create a CompiledKernel for all the ops in the function, except Parameters and Results.
+    NodeVector ck_ops;
+    for (auto op : func->get_ordered_ops())
+    {
+        // All ops must be supported by MLIR compiler
+        if (!is_supported_mlir_op(op))
+            return false;
+
+        if (TI(Parameter) != TI(*op) && TI(Result) != TI(*op))
+        {
+            ck_ops.push_back(op);
+        }
+    }
+
+    NodeVector ck_args;
+    for (auto& param : func->get_parameters())
+    {
+        ck_args.push_back(param);
+    }
+
+    NodeVector ck_outputs = std::move(get_subgraph_outputs(ck_ops, {} /*exclusions*/));
+    if (ck_outputs.size() != 1)
+        return false;
```

Comment:
curly braces around code block

Reply:
Fixed this file and compiler.cpp. However, for other files we follow MLIR coding style since we extend their C++ classes and code.